gremlin-scala/src/main/scala/gremlin/scala/ScalaGraph.scala

```diff
@@ -6,21 +6,25 @@ import shapeless._
 
 case class ScalaGraph[G <: Graph](graph: G) {
 
-  def addVertex() = ScalaVertex(graph.addVertex())
+  def addVertex(): ScalaVertex = graph.addVertex()
 
-  def addVertex(label: String) = ScalaVertex(graph.addVertex(label))
+  def addVertex(label: String): ScalaVertex = graph.addVertex(label)
 
   def addVertex(properties: Map[String, Any]): ScalaVertex = {
-    val v = addVertex()
-    v.setProperties(properties)
-    v
+    graph.addVertex(properties.toSeq.flatMap {
+      case (k, v) =>
+      val label = T.label.name
+      val id = T.id.name
+      k match {
+        case `label` => Seq(T.label, v.asInstanceOf[AnyRef])
+        case `id` => Seq(T.id, v.asInstanceOf[AnyRef])
+        case _ => Seq(k, v.asInstanceOf[AnyRef])
+      }
+    }: _*)
   }
```

Comment:
The syntax to make this work is actually:
```
import org.apache.tinkerpop.gremlin.structure.T
import shapeless.|∨|

def addVertex[K: (T |∨| String)#λ](p: Map[K, Any]) = p

addVertex(Map(T.label -> "test", T.id -> 9, "name" -> "value"))
```
which does not actually run.  From `console` we get:
```
scala> addVertex(Map(T.label -> "test", T.id -> 9, "name" -> "value"))
<console>:24: error: Cannot prove that (Comparable[_ >: String with org.apache.tinkerpop.gremlin.structure.T <: Comparable[_ >: String with org.apache.tinkerpop.gremlin.structure.T <: java.io.Serializable] with java.io.Serializable] with java.io.Serializable => Nothing) => Nothing <:< org.apache.tinkerpop.gremlin.structure.T => Nothing with String => Nothing => Nothing.
       addVertex(Map(T.label -> "test", T.id -> 9, "name" -> "value"))
 ```
Any advise here?